medusa/providers/torrent/html/bjshare.py

```diff
@@ -0,0 +1,470 @@
+                        continue
+
+                    details = cells[labels.index("Nome")].find_all("font")
+                    if len(details) >= 3:
+                        resolution = self.quality[details[2].get_text()]
+                        resolution = details[6] if resolution is None else resolution
+                    else:
+                        resolution = '480p'
+
+                    source = '' if not quality_re else quality_re.group(0)
+                    codec = ''
+                    audio = '' if not audio_re else audio_re.group(0)
+                    ext = '' if not format_re else format_re.group(0)
+
+                    torrent_size = size_re.group(0)
+                    size = convert_size(torrent_size, units=self.units) or -1
+                    try:
+                        season = re.findall(r"(?:s|season)(\d{2})", title, re.I)[0]
+                    except IndexError:
+                        season = ''
+
+                    try:
+                        episode = re.findall(r"(?:e|x|episode|ep|\n)(\d{2,4})", title, re.I)[0]
+                    except IndexError:
+                        episode = ''
+
+                    title = torrent['international_name']
+                    if not torrent['national_title'] and season.isdigit() and episode.isdigit():
+                        title = title[:title.rfind('-')].strip()
+
+                    if anime and title in self.animes_with_broken_seasons_numbering:
+                        # Some animes season is broken, so ignore it and include only the episode
+                        # In this case, the episode is in absolute format
+                        torrent_name = u"{0} E{1} {2} {3} {4} {5}" \
+                            .format(title, episode, resolution, source, codec, audio)
+                    elif season.isdigit() and not episode.isdigit():
+                        # Season Pack
+                        torrent_name = u"{0} S{1} {2} {3} {4} {5}" \
+                            .format(title, season, resolution, source, codec, audio)
+                    elif season.isdigit() and episode.isdigit():
+                        # Season with episode included
+                        torrent_name = u"{0} S{1}E{2} {3} {4} {5} {6}" \
+                            .format(title, season, episode, resolution, source, codec, audio)
+                    else:
+                        # Probably movie or something else that does not have season or episode
+                        torrent_name = u"{0} {1} {2} {3} {4}" \
+                            .format(title, resolution, source, codec, audio)
+                    # Removing accents due to an bug that i noticed that the torrent is not being sent to
+                    # qBittorrent when accents are present, there was no error in log.
+                    torrent_name = re.sub('\s+', ' ', self._remove_accents(torrent_name)).strip()
```

Comment:
Accents shouldn't removed, the string should left intact. All it should be sending to qBittorrent is the torrent url if I'm not mistaken, so this may sound like a locale config.

Reply:
So we need to figure out why it was not sending to torrent client when accent was present.. I'll do some more tests

Reply:
Tested again, and i receive the telegram message just fine saying about the download, but its not added to qbittorrent..

That's the response from qbittorrent when i try to send an torrent with accent:

```2018-05-18 13:04:06 INFO     SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] Beginning to manual snatch release: Black.Clover.-.S01E32.MKV./.H.264./.AAC./.HDTV./.Legendado./.Japonês./.1080p
2018-05-18 13:04:06 INFO     SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] Downloading Black.Clover.-.S01E32.MKV./.H.264./.AAC./.HDTV./.Legendado./.Japonês./.1080p with 8 seeders and 0 leechers and size 547.08 MB from BJ-Share
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] GET URL: https://bj-share.me/torrents.php?action=download&id=153167&authkey=a04a6d622a11a69cacc0bc6180c14c13&torrent_pass=n8y6e8lkljrp78gixy4651vutqueo2pg [Status: 200]
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] User-Agent: Medusa/0.2.2 (Windows; 10; cdb05a61-5ab4-11e8-92f1-7cdd9083bde4)
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] Calling qbittorrent Client
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] GET URL: http://localhost:9496/version/api [Status: 200]
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] User-Agent: Medusa/0.2.2 (Windows; 10; cdb05a61-5ab4-11e8-92f1-7cdd9083bde4)
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] POST URL: http://localhost:9496/login [Status: 200]
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] User-Agent: Medusa/0.2.2 (Windows; 10; cdb05a61-5ab4-11e8-92f1-7cdd9083bde4)
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] With post data: username=**********&password=**********
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] qbittorrent: Requested a POST connection to http://localhost:9496/command/upload with params: None Data: None
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] POST URL: http://localhost:9496/command/upload [Status: 200]
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] User-Agent: Medusa/0.2.2 (Windows; 10; cdb05a61-5ab4-11e8-92f1-7cdd9083bde4)
 Content-Disposition: form-data; name="torrents"; filename*=utf-...
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] qbittorrent: Response to POST request is 
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] GET URL: http://localhost:9496/version/api [Status: 200]
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] User-Agent: Medusa/0.2.2 (Windows; 10; cdb05a61-5ab4-11e8-92f1-7cdd9083bde4)
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] qbittorrent: Requested a POST connection to http://localhost:9496/version/api with params: None Data: {u'hash': '2b44e208ac311fae598013251bb7bfbd9e16ba9d'}
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] POST URL: http://localhost:9496/version/api [Status: 200]
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] User-Agent: Medusa/0.2.2 (Windows; 10; cdb05a61-5ab4-11e8-92f1-7cdd9083bde4)
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] With post data: hash=2b44e208ac311fae598013251bb7bfbd9e16ba9d
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] qbittorrent: Response to POST request is 17
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] GET URL: http://localhost:9496/version/api [Status: 200]
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] User-Agent: Medusa/0.2.2 (Windows; 10; cdb05a61-5ab4-11e8-92f1-7cdd9083bde4)
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] GET URL: http://localhost:9496/version/api [Status: 200]
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] User-Agent: Medusa/0.2.2 (Windows; 10; cdb05a61-5ab4-11e8-92f1-7cdd9083bde4)
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] qbittorrent: Requested a POST connection to http://localhost:9496/command/setCategory with params: None Data: {u'category': u'medusa', u'hashes': '2b44e208ac311fae598013251bb7bfbd9e16ba9d'}
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] POST URL: http://localhost:9496/command/setCategory [Status: 200]
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] User-Agent: Medusa/0.2.2 (Windows; 10; cdb05a61-5ab4-11e8-92f1-7cdd9083bde4)
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] With post data: category=medusa&hashes=2b44e208ac311fae598013251bb7bfbd9e16ba9d
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] qbittorrent: Response to POST request is 
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] qbittorrent: Requested a POST connection to http://localhost:9496/command/increasePrio with params: None Data: {u'hashes': '2b44e208ac311fae598013251bb7bfbd9e16ba9d'}
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] POST URL: http://localhost:9496/command/increasePrio [Status: 200]
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] User-Agent: Medusa/0.2.2 (Windows; 10; cdb05a61-5ab4-11e8-92f1-7cdd9083bde4)
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] With post data: hashes=2b44e208ac311fae598013251bb7bfbd9e16ba9d
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] qbittorrent: Response to POST request is 
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] Checking if we should stop searching for a better quality for for episode Black.Clover.-.S01E32.MKV./.H.264./.AAC./.HDTV./.Legendado./.Japonês./.1080p
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] Matched release G:\Meu Drive\Media\Animes\Black Clover\Season 1\Black.Clover.S01E32.032.1080p.HDTV-Multiple Subtitle to a series in your database: Black Clover
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] Parsed G:\Meu Drive\Media\Animes\Black Clover\Season 1\Black.Clover.S01E32.032.1080p.HDTV-Multiple Subtitle into title: Black Clover, season: 1, episode: [32], episode_title: 032, screen_size: 1080p, format: HDTV, release_group: Multiple Subtitle, type: episode, parsing_time: 0.169000148773, total_time: 0.169000148773, absolute_episode: [], quality: 1080p HDTV
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] Found codec for Black Clover Three-Leaf Sprouts
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] 331753: Episode has no release group, replacing it with Multiple Subtitle
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] 331753: Formatting pattern: %SN - %Sx32 - 032 - %EN - %QN -> Black Clover - 1x32 - 032 - Three-Leaf Sprouts - 1080p HDTV
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] Sending a Telegram message for Black Clover - 1x32 - 032 - Three-Leaf Sprouts - 1080p HDTV with 8 seeders and 0 leechers from BJ-Share
2018-05-18 13:04:07 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] Telegram in use with API KEY: **********
2018-05-18 13:04:08 INFO     SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] Telegram message sent successfully.
2018-05-18 13:04:08 DEBUG    SNATCHQUEUE-MANUALSNATCH-331753 :: [71edd70] Transaction with 1 queries executed```

Reply:
Tested it without slashes, and same happens, it will not send to qbittorrent (but if i remove the accents, it send just fine, so it must be something with accents or encoding)...
Tested it to blackhole, and it work fine even with accents, so must be something with the api call to qbittorrent...

Reply:
Ok, please open an issue for that then. That's an issue with qBittorrent and affects all providers with non ascii results. For testing provider for now, you can use blackhole.

Reply:
Ok, i'll open an issue, i did an big refactor to the code, if you can take a look if it's fine now..
(i'll need to do more tests to be sure that everything is working, i did just few tests)